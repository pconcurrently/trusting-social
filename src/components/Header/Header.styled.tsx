import styled from 'styled-components';

import { IApplicationTheme } from '../../shared/themes';

interface ITheme {
  theme: IApplicationTheme;
}

export const StyledHeader = styled.header`
  width: 100%;
  height: 50px;
  display: flex;
  justify-content: space-between;
  font-weight: 600;
  font-size: 14px;
  position: relative;
  z-index: 1;

  ul {
    display: table;
    height: 50px;
    margin: 0;

    li {
      display: table-cell;
      list-style: none;
      vertical-align: middle;
      transition: 0.2s;

      &:hover:not(:first-child) {
        a {
          border-bottom: solid 2px #fff;
        }
      }

      a {
        display: inline-block;
        text-decoration: none;
        height: 50px;
        line-height: 50px;
        padding: 0 9px;
        margin: 0 9px;
        color: #fff;
        border-bottom: solid 2px transparent;
      }
    }
  }
`;
