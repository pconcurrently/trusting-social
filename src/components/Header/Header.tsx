import React from 'react';
import styled from 'styled-components';

const StyledHeader = styled.header`
    background-color: ${({ theme }) => theme.PRIMARY};
    height: 100px;
`;

interface IProps {
    switchTheme: () => void;
}

export const Header: React.SFC<IProps> = ({ switchTheme }) => {
    return (
        <StyledHeader>
            <button onClick={switchTheme}>Change theme</button>
        </StyledHeader>
    );
}
