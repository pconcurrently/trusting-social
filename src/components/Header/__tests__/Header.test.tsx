import React from 'react';
import { mount, shallow } from 'enzyme';
import { Header } from '../Header';

describe('App', () => {
  const switchTheme = jest.fn();

  it('should render Header', () => {
    const container = shallow(<Header switchTheme={switchTheme} />);
    expect(container).toMatchSnapshot();
  });

  it('should fire switchTheme function', () => {
    const container = mount(<Header switchTheme={switchTheme} />);
    const event = { currentTarget: { checked: true } };
    expect(container.find('input#checkbox').props().checked).toBeFalsy();
    container.find('input#checkbox').simulate('change', event);
    expect(switchTheme).toBeCalled();
  });
});
