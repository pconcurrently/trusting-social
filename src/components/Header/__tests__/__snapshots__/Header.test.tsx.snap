// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
"<styled.header>
  <div className=\\"logo-wrapper\\">
    <img src=\\"airbnb-logo.svg\\" alt=\\"Logo\\" />
  </div>
  <ul>
    <li>
      <ThemeSwitcher onClick={[Function: mockConstructor]} />
    </li>
    <li>
      <a href=\\"#\\">
        Become a host
      </a>
    </li>
    <li>
      <a href=\\"#\\">
        Help
      </a>
    </li>
    <li>
      <a href=\\"#\\">
        Sign up
      </a>
    </li>
    <li>
      <a href=\\"#\\">
        Login
      </a>
    </li>
  </ul>
</styled.header>"
`;
